src/code/compareSimulations.R

```
@@ -0,0 +1,498 @@
+    simName1 = splitSimNameFromPath(anyParam_1_path)$SimName;
+    
+    anyParam_1_path = "";
+  }
+  else{
+    simName1 = "Not available";
+  }
+  #Name of the second simulation
+  if (length(DCI_Info2$ReferenceTab$AllParameters$Path) > 0){
+    anyParam_2_path = DCI_Info2$ReferenceTab$AllParameters$Path[1];
+    simName2 = splitSimNameFromPath(anyParam_2_path)$SimName;
+    
+    anyParam_2_path = "";
+  }
+  else{
+    simName2 = "Not available";
+  }
+  #Store simulation names into the output.
+  results$SimNames = c(simName1, simName2);
+  
+  #If same DCI is compared, return no differences without comparison
+  if (DCI_Info1$Handle == DCI_Info2$Handle){
+    objects_diff = list();
+    objects_diff$Path = c();
+    objects_diff$isExistingFirst = c();
+    objects_diff$isExistingSecond = c();
+    objects_diff$ValueFirst = c();
+    objects_diff$ValueSecond = c();
+    objects_diff$IsFormulaFirst = c();
+    objects_diff$IsFormulaSecond = c();
+    objects_diff$FormulaFrist = c();
+    objects_diff$FormulaSecond = c();
+    results$Params_diff = objects_diff;
+    results$Species_diff = objects_diff;
+    results$Observer_diff = objects_diff;
+    results$TimeIdentical = TRUE;
+    
+    #If a path to output file a specified, write the results into a CSV-file.
+    if (!resultsCSV_path == "")
+    {
+      writeToCSV(CSV_path = resultsCSV_path, results = results);
+    }
+    return(results)
+  }
+  
+  #Compare parameters, species, and observers.
+  params_diff = compareStructures("Parameter", DCI_Info1, DCI_Info2, numTolerance = numTolerance, simName1 = simName1, simName2 = simName2, ignoreFormula = ignoreFormula);
+  species_diff = compareStructures("Species", DCI_Info1, DCI_Info2, numTolerance = numTolerance, simName1 = simName1, simName2 = simName2, ignoreFormula = ignoreFormula);
+  observer_diff = compareStructures("Observer", DCI_Info1, DCI_Info2,numTolerance = numTolerance, simName1 = simName1, simName2 = simName2, ignoreFormula = ignoreFormula);
```

do you have to name all parameters explicitely? (Eg. numTolerance = numTolerance, simName1 = simNAme1 etc..)
t

---

You mean, if one could write

```params_diff = compareStructures("Parameter", DCI_Info1, DCI_Info2, numTolerance, simName1, simName2, ignoreFormula);```
? Yes, this would work. I just made it my habit explicitely naming everything, since the order of parameters is not always clear and I had some weird bugs that were really hard to track, just because I passed the parameters in a wrong order and R just silently accepted it (weak typed).

---

That make sense. But then why not name "Parameter", DCI_Info1 etc.. as well ?
It's always suspicious for me when method have 8 parameters. I am not very familiar with R but typically what we would no in other languages (static or dynamic) is to create an object that would be the parameter
sthg like 
```
{
 "type": Parameter,
 "numTolerance" numTolerance
...
}
```
and pass that object to the method. It your case, it would remove A LOT of the duplicated code.

I am not sure if this is something done in R?

---

OOP in R is possible (somehow) but I have no experience with it. I can pack all parameters in a named list (basically a map) and pass it to the methods. Will try it out.

---

It's more functional than OOP. Typically done in javascript where the amount of parameters tends to explode. 

If this is not something done in R in general, forget about it. I don't know enough R to make really good suggestions :)